packages/holocron/src/loadModule.web.js

```diff
@@ -13,54 +13,84 @@
-  return new Promise((resolve, reject) => {
-    if (typeof moduleName !== 'string') {
-      throw new TypeError('loadModule: moduleName must be a string');
-    }
+function createScript({ url, integrity, onLoad = noop }) {
+  const script = global.document.createElement('script');
+  const isProduction = process.env.NODE_ENV === 'production';
+  const head = global.document.getElementsByTagName('head')[0];
 
-    if (typeof moduleData !== 'object') {
-      throw new TypeError('loadModule: moduleData must be an object');
-    }
+  script.type = 'text/javascript';
+  script.async = true;
+  script.crossOrigin = 'anonymous';
 
-    // eslint-disable-next-line no-underscore-dangle
-    const integrity = moduleData.getIn([window.__holocron_module_bundle_type__, 'integrity']);
-    // eslint-disable-next-line no-underscore-dangle
-    const url = moduleData.getIn([window.__holocron_module_bundle_type__, 'url']);
-    const isProduction = process.env.NODE_ENV === 'production';
-
-    const head = global.document.getElementsByTagName('head')[0];
-    const script = global.document.createElement('script');
-    script.type = 'text/javascript';
-    script.charset = 'utf-8';
-    script.async = true;
-    script.timeout = 120000;
-
-    script.crossOrigin = 'anonymous';
-
-    if (isProduction) {
-      script.integrity = integrity;
-    }
-    const clientCacheRevision = getModuleMap().get('clientCacheRevision', getModuleMap().get('key'));
-    script.src = isProduction && clientCacheRevision ? `${url}?clientCacheRevision=${clientCacheRevision}` : url;
-    const timeout = setTimeout(onScriptComplete, 120000);
+  if (isProduction && integrity) {
+    script.integrity = integrity;
+  }
+
+  const clientCacheRevision = getModuleMap().get(
+    'clientCacheRevision',
+    getModuleMap().get('key')
+  );
+  script.src = isProduction && clientCacheRevision
+    ? `${url}?clientCacheRevision=${clientCacheRevision}`
+    : url;
+
+  const listener = new Promise((resolve, reject) => {
     script.addEventListener('error', (event) => {
-      clearTimeout(timeout);
       reject(event.message);
     });
 
     script.addEventListener('load', () => {
-      clearTimeout(timeout);
-      resolve(getModule(moduleName));
+      resolve(onLoad());
     });
+  });
+
+  head.appendChild(script);
+
+  return listener;
+}
 
-    head.appendChild(script);
+function loadModuleFallbackExternals(moduleName) {
+  const fallbacks = getUnregisteredRequiredExternals(moduleName);
+  const baseUrl = getModuleMap().getIn(['modules', moduleName, 'baseUrl']);
+  return fallbacks.map(({ filename, integrity }) => createScript({
+    url: `${baseUrl}${filename}`,
+    integrity,
+  }));
+}
+
+async function loadModule(moduleName, moduleData) {
+  // first load all module fallbacks.
+  // possible race condition if two modules require the same fallback and version.
+  await Promise.all(loadModuleFallbackExternals(moduleName));
```

Comment:
Can we return the `Promise.all` from within the `loadModuleFallbackExternals` function instead? that way this line becomes
```suggestion
  await loadModuleFallbackExternals(moduleName)
```

Assuming that `loadModuleFallbackExternals` is only used here